casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -1,29 +1,33 @@
 package coop.rchain.casper
 
+import cats.implicits._
 import com.google.protobuf.ByteString
 import coop.rchain.casper.protocol.BlockMessage
 import coop.rchain.casper.util.DagOperations
 import coop.rchain.casper.util.ProtoUtil.{mainParent, parents, weightMap}
 
 import scala.annotation.tailrec
-import scala.collection.mutable
 import scala.collection.immutable.{HashMap, HashSet}
 
+import coop.rchain.catscontrib.ListContrib
+
 object Estimator {
   type BlockHash = ByteString
   type Validator = ByteString
 
-  private val decreasingOrder = Ordering[Int].reverse
+  implicit val decreasingOrder = Ordering[Int].reverse
 
   def tips(childMap: HashMap[BlockHash, HashSet[BlockHash]],
            blockLookup: HashMap[BlockHash, BlockMessage],
            latestMessages: HashMap[Validator, BlockHash],
            genesis: BlockMessage): IndexedSeq[BlockMessage] = {
     @tailrec
-    def sortChildren[A <: (BlockHash) => Int](blocks: IndexedSeq[BlockHash],
-                                              scores: A): IndexedSeq[BlockHash] = {
+    def sortChildren(blocks: IndexedSeq[BlockHash],
+                     scores: collection.Map[BlockHash, Int]): IndexedSeq[BlockHash] = {
```

Comment:
It's probably fine to just have `Map` instead of `collection.Map` since you are only ever dealing with immutable Maps.

Reply:
I'll push this to the other PR.